File: app/src/main/java/com/levibostian/androidblanky/adapter/recyclerview/AboutListAdapter.java
package com.levibostian.androidblanky.adapter.recyclerview;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.levibostian.androidblanky.R;

import java.util.ArrayList;

public class AboutListAdapter extends RecyclerView.Adapter<AboutListAdapter.ViewHolder> {

    private ArrayList<String> mData;
    private ItemClickListener mListener;

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private TextView mListTitle;

        public ViewHolder(View view) {
            super(view);

            mListTitle = (TextView) view.findViewById(R.id.about_list_title_textview);
        }
    }

    public interface ItemClickListener {
        void onItemClick(int position);
    }

    public AboutListAdapter(ArrayList<String> data, ItemClickListener listener) {
        mData = data;
        mListener = listener;
    }

    @Override
    public AboutListAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.about_list_row, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position) {
        holder.mListTitle.setText(mData.get(position));
        holder.mListTitle.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mListener.onItemClick(holder.getAdapterPosition());
            }
        });
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

}

File: app/src/main/java/com/levibostian/androidblanky/adapter/recyclerview/AboutListAdapter.kt
package com.levibostian.androidblanky.adapter.recyclerview

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.levibostian.androidblanky.R
import java.util.*

class AboutListAdapter(private val mData: ArrayList<String>, private val mListener: ItemClickListener) : RecyclerView.Adapter<AboutListAdapter.ViewHolder>() {

    interface ItemClickListener {
        fun onItemClick(position: Int)
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        val mListTitle: TextView

        init {
            mListTitle = view.findViewById(R.id.about_list_title_textview) as TextView
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        val view: View = LayoutInflater.from(parent!!.context).inflate(R.layout.about_list_row, parent, false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
        holder!!.mListTitle.text = mData[position]
        holder.mListTitle.setOnClickListener {
            mListener.onItemClick(holder.adapterPosition)
        }
    }

    override fun getItemCount(): Int {
        return mData.size
    }

}